linalg/sundials.hpp

```diff
@@ -23,384 +23,433 @@
-                        CV_NEWTON (usually with CV_BDF).
-       For parameter desciption, see the CVodeCreate documentation (cvode.h). */
-   CVODESolver(int lmm, int iter);
+   /// Construct a serial wrapper to SUNDIALS' CVODE integrator.
+   /** @param[in] lmm Specifies the linear multistep method, the options are:
+                      - CV_ADAMS - implicit methods for non-stiff systems,
+                      - CV_BDF   - implicit methods for stiff systems. */
+   CVODESolver(int lmm);
 
 #ifdef MFEM_USE_MPI
-   /// Construct a parallel CVODESolver, a wrapper for SUNDIALS' CVODE solver.
-   /** @param[in] comm  The MPI communicator used to partition the ODE system.
-       @param[in] lmm   Specifies the linear multistep method, the options are
-                        CV_ADAMS (explicit methods) or CV_BDF (implicit
-                        methods).
-       @param[in] iter  Specifies type of nonlinear solver iteration, the
-                        options are CV_FUNCTIONAL (usually with CV_ADAMS) or
-                        CV_NEWTON (usually with CV_BDF).
-       For parameter desciption, see the CVodeCreate documentation (cvode.h). */
-   CVODESolver(MPI_Comm comm, int lmm, int iter);
+   /// Construct a parallel wrapper to SUNDIALS' CVODE integrator.
+   /** @param[in] comm The MPI communicator used to partition the ODE system
+       @param[in] lmm  Specifies the linear multistep method, the options are:
+                       - CV_ADAMS - implicit methods for non-stiff systems,
+                       - CV_BDF   - implicit methods for stiff systems. */
+   CVODESolver(MPI_Comm comm, int lmm);
 #endif
 
+   /** @brief Initialize CVODE: calls CVodeCreate() to create the CVODE
+       memory and set some defaults.
+
+       If the CVODE memory has already been created, it checks if the problem
+       size has changed since the last call to Init(). If the problem is the
+       same then CVodeReInit() will be called in the next call to Step(). If
+       the problem size has changed, the CVODE memory is freed and realloced
+       for the new problem size. */
+   /** @param[in] f_ The TimeDependentOperator that defines the ODE system.
+
+       @note All other methods must be called after Init(). */
+   void Init(TimeDependentOperator &f_);
+
+   /// Integrate the ODE with CVODE using the specified step mode.
+   /** @param[in,out] x  On output, the solution vector at the requested output
+                         time tout = @a t + @a dt.
+       @param[in,out] t  On output, the output time reached.
+       @param[in,out] dt On output, the last time step taken.
+
+       @note On input, the values of @a t and @a dt are used to compute desired
+       output time for the integration, tout = @a t + @a dt.
+   */
+   virtual void Step(Vector &x, double &t, double &dt);
+
+   /// Attach an MFEM linear solver to CVODE.
+   void UseMFEMLinearSolver();
+
+   /// Attach SUNDIALS linear solver to CVODE.
+   void UseSundialsLinearSolver();
```

Comment:
In the above two methods (`UseMFEMLinearSolver` and `UseSundialsLinearSolver`), where is the solver that being set? It will be nice to clarify this in the doxygen documentation.